tangram_bundler/__init__.py

```diff
@@ -1,100 +1,233 @@
+
+def resolveSceneUrls(yamlRoot, filename):
+    basePath = filename
+    if 'textures' in yamlRoot:
+        resolveSceneTextureUrls(yamlRoot['textures'], basePath)
+    if 'styles' in yamlRoot:
+        resolveSceneStyleUrls(yamlRoot['styles'], basePath)
+    if 'fonts' in yamlRoot:
+        resolveSceneFontsUrl(yamlRoot['fonts'], basePath)
+
+
+def importSceneRecursive(yamlRoot, filename, allImports, importedScenes):
+    if filename in importedScenes:
+        return
+    importedScenes.add(filename)
+    sceneNode = allImports[filename]
+    if ((sceneNode is None) or (type(sceneNode) is not dict)):
+        return
+    imports = getSceneImports(filename)
+    sceneNode['import'] = None
+    for importScene in imports:
+        importSceneRecursive(yamlRoot, importScene, allImports, importedScenes)
+    importedScenes.remove(filename)
+    print "merging scene file to root: ", filename
+    mergeMapFields(yamlRoot, sceneNode)
+    resolveSceneUrls(yamlRoot, filename)
 
 # ================================== Main functions
-def bundler(full_filename):
-    print full_filename, os.getcwd()
-    filename, file_extension = os.path.splitext(full_filename)
-    if file_extension == '.yaml':
-        all_dependencies = []
-
-        # 1st order dependencies
-        addDependencies(all_dependencies, './'+full_filename)
-
-        # 2nd order theme dependencies
-        try:
-            for file in os.listdir(os.getcwd() + "/themes"):
-                if file.endswith(".yaml"):
-                    # track like normal
-                    all_dependencies.append("themes/" + file)
-                    # some themes require additional assets
-                    print "looking at dependencies for: %s" % (file,)
-                    addDependencies(all_dependencies, "themes/" + file)
-        except Exception:
-            print "\tskipping: themes (none found)"
-
-        files = list(set(all_dependencies))
-
-        print "Bundling ",filename,"width",len(files),"dependencies, into ",filename+".zip"
-        zipf = zipfile.ZipFile(filename+'.zip', 'w', zipfile.ZIP_DEFLATED)
-        for file in files:
+def bundler(filename):
+    print filename, os.getcwd()
+
+    allImports = {}
+    absFilename = os.path.abspath(filename)
+    allImports[absFilename] = loadYaml(absFilename)
+    collectAllImports(allImports)
+    
+    rootNode = {}
+    importedScenes = set()
+    importSceneRecursive(rootNode, absFilename, allImports, importedScenes)
+
+    allDependencies = []
+    basePath = os.path.dirname(absFilename)
+    fetchDependencies(allDependencies, rootNode, basePath)
+
+    for file in allImports:
+        allDependencies.append(os.path.relpath(file, basePath))
+
+    files = list(set(allDependencies))
+    print files
+
+    print "Bundling ",filename,"width",len(files),"dependencies, into ",filename+".zip"
+    zipf = zipfile.ZipFile(filename+'.zip', 'w', zipfile.ZIP_DEFLATED)
```

Comment:
`filename` here needs to strip off the extension and then add zip

Reply:
Ahh, I did not realize usage of extension was also to get the final zip file name. Will fix.

Reply:
Should be fixed by 49aff91